plugins/pencilblue/controllers/admin/content/navigation/nav_item_form.js

```diff
@@ -20,147 +20,159 @@ var async = require('async');
+
+  /**
+   * Interface for creating and editing navigation items
+   * @class NavItemFormController
+   */
+  function NavItemFormController(){
+    this.navItem = null;
+  }
+  util.inherits(NavItemFormController, pb.BaseController);
+
+  //statics
+  var SUB_NAV_KEY = 'nav_item_form';
+
+  NavItemFormController.prototype.render = function(cb) {
+    var self = this;
+    var vars = this.pathVars;
+
+    //gather all data
+    this.gatherData(vars, function(err, data) {
+      if (util.isError(err)) {
+        throw err;
+      }
+      else if(!data.navItem) {
+        self.reqHandler.serve404();
+        return;
+      }
+
+      self.navItem = data.navItem;
+      var contentSearchValue = self.navItem.contentSearchValue ? self.navItem.contentSearchValue.toString() : '';
+      delete self.navItem.contentSearchValue;
+
+      data.pills = pb.AdminSubnavService.get(self.getSubnavKey(), self.ls, self.getSubnavKey(), self.navItem);
+      var angularObjects = pb.ClientJs.getAngularObjects(data);
+
+      self.setPageName(self.navItem[pb.DAO.getIdField()] ? self.navItem.name : self.ls.get('NEW_NAV_ITEM'));
+      self.ts.registerLocal('angular_objects', new pb.TemplateValue(angularObjects, false));
+      self.ts.registerLocal('content_type', '{{section.type}}');
+      self.ts.registerLocal('selection_id_field', 'item');
+      self.ts.load('admin/content/navigation/nav_item_form', function(err, result) {
+        cb({content: result});
+      });
+    });
+  };
+
+  NavItemFormController.prototype.gatherData = function(vars, cb) {
+    var self = this;
+    var dao = new pb.DAO();
+
+    var tasks = {
+
+      //get editors
+      editors: function(callback) {
+        pb.users.getWriterOrEditorSelectList(self.session.authentication.user_id, false, callback);
+      },
+
+      //get parents
+      parents: function(callback) {
+        var sectionService = new pb.SectionService();
+        sectionService.getParentSelectList(self.pathVars.id, function(err, parents) {
+          if(util.isError(err)) {
+            callback(err, parents);
+            return;
+          }
+
+          parents.unshift({_id: '', name: ''});
+          callback(null, parents);
         });
-    };
+      },
+
+      //form tabs
+      tabs: function(callback) {
+        var tabs = [
+          {
+            active: 'active',
+            href: '#section_settings',
+            icon: 'cog',
+            title: self.ls.get('SETTINGS')
+          }
+        ];
+        callback(null, tabs);
+      },
+
+      navigation: function(callback) {
+        callback(null, pb.AdminNavigation.get(self.session, ['content', 'sections'], self.ls));
+      },
+
+      types: function(callback) {
+        callback(null, SectionService.getTypes(self.ls));
+      },
+
+      articles: function(callback) {
+        dao.q('article', {where: {draft: {$ne: 1}}}, function(err, articles) {
+          callback(err, articles);
+        });
+      },
```

Comment:
I really don't like this.  This is a step in the wrong direction no matter how little traffic the page gets.  It is still a burden on the DB to return data for a page that only shows the data if a specific selection is made.  If we need an API endpoint to retrieve and load all articles/pages we can do it.  That is a much better stop gap solution until a better UX can be provided.